### PR TITLE
Add Scalyr logger

### PR DESCRIPTION
**Pull request by pieterlexis-tomtom**
This PR adds support for sending DNSTap logs to [Scalyr](https://www.dataset.com/) (they have rebranded to DataSet some time ago). Because of some limits in Scalyr, this PR also adds a special (although CPU intensive) output mode that flattens the JSON to one level of key-value pairs (see loggers.md for more info). I have done some testing locally by sending the logs to Scalyr and it seems to work just fine.

This PR is a draft because I want to test this in a real-world deployment first and get some (early) feedback on the code.

I was also wondering if it would make sense to make the flat JSON a transformer?

### CODE REVIEW

**Comment by dmachard** [comment]
Thanks for your contribution
I will take look later but I like your idea of flat JSON

**Comment by dmachard** [comment]
lgtm

As said before, flat JSON is a good idea even for other loggers too.
Perhaps the `DnsMessage` struct can be updated with a new function `ToFlatJson` ?

**Comment by pieterlexis-tomtom** [comment]
I added a `Flatten` function to `DnsMessage` that returns `map[string]interface{}` so extra data can be added :). I also added a `panic` when the routing config is wrong (which cost me a day of debugging) like:

```yaml
routing:
  - from: foo
  - to: bar
```

File: doc/loggers.md
Comment (dmachard) [inline doc/loggers.md:1]: I was wondering if the "## Flat JSON export format" chapter can be moved in doc/dnsjson.md ?

**Comment by dmachard** [comment]
I was wondering if the "## Flat JSON export format" documentation can be moved in doc/dnsjson.md ?

**Review by dmachard — COMMENTED** [review]
I was wondering if the "## Flat JSON export format" documentation can be moved in doc/dnsjson.md ?

**Comment by pieterlexis-tomtom** [comment]
> I was wondering if the "## Flat JSON export format" documentation can be moved in doc/dnsjson.md ?

makes sense, done!

**Review by dmachard — APPROVED** [review]

**Pull request merged**